db/indextest/database_test.go

```diff
@@ -0,0 +1,175 @@
+/*
```

Comment:
It feels like these tests could also live in indextest_test.go - any reason you think they need their own file?

Reply:
I created `db/indextest/database_test.go` just because these test lived in `db/database_test.go`. I don't have any strong reason other than this.